go/vt/vttablet/tabletmanager/vreplication/vplayer_flaky_test.go

```diff
@@ -2802,6 +2804,154 @@ func TestPlayerInvalidDates(t *testing.T) {
+		output: "update t1 set val1='bbb', id2=99, val2='CCC' where id=1",
+		table:  "t1",
+		data: [][]string{
+			{"1", "bbb", "blb1", "99", "blb22", "CCC"},
+		},
+	}, { // 5. PartialQueryTemplate-Update=2, PartialQueryCount-Update=4 (blb1 is not updated, same as #1)
+		input:  "update t1 set blb2 = 'blb21' where id = 1",
+		output: "update t1 set val1='bbb', id2=99, blb2='blb21', val2='CCC' where id=1",
+
+		table: "t1",
+		data: [][]string{
+			{"1", "bbb", "blb1", "99", "blb21", "CCC"},
+		},
+	}, { // 6. Not a partial update
+		input:  "update t1 set blb2 = 'blb222', blb1 = 'blb11' where id = 1",
+		output: "update t1 set val1='bbb', blb1='blb11', id2=99, blb2='blb222', val2='CCC' where id=1",
+		table:  "t1",
+		data: [][]string{
+			{"1", "bbb", "blb11", "99", "blb222", "CCC"},
+		},
+	}, { // 7. PartialQueryTemplate-Insert=2, PartialQueryCount-Insert=2 (txt1 is not inserted)
+		input:  "insert into t2(id,val1,id2,val2) values (1,'aaa',10,'AAA')",
+		output: "insert into t2(id,val1,id2,val2) values (1,'aaa',10,'AAA')",
+		table:  "t2",
+		data: [][]string{
+			{"1", "aaa", "", "10", "AAA"},
+		},
+	}, { // 7. PartialQueryTemplate-Insert=2, PartialQueryCount-Insert=3 (txt1 is not inserted, same as #7)
+		input:  "insert into t2(id,val1,id2,val2) values (1,'bbb',20,'BBB')",
+		output: "insert into t2(id,val1,id2,val2) values (1,'bbb',20,'BBB')",
+		table:  "t2",
+		data: [][]string{
+			{"1", "aaa", "", "10", "AAA"},
+			{"1", "bbb", "", "20", "BBB"},
+		},
+	}, { // 8. Not a partial update, all columns are present
+		input:  "update t2 set txt1 = 'txt1' where id = 1 and val1 = 'aaa'",
+		output: "update t2 set txt1='txt1', id2=10, val2='AAA' where id=1 and val1='aaa'",
+		table:  "t2",
+		data: [][]string{
+			{"1", "aaa", "txt1", "10", "AAA"},
+			{"1", "bbb", "", "20", "BBB"},
+		},
+	}, { // 9. Not a partial update, all columns are present, same as #8
+		input:  "update t2 set val2 = 'DDD', txt1 = 'txt2' where id = 1 and val1 = 'bbb'",
+		output: "update t2 set txt1='txt2', id2=20, val2='DDD' where id=1 and val1='bbb'",
+		table:  "t2",
+		data: [][]string{
+			{"1", "aaa", "txt1", "10", "AAA"},
+			{"1", "bbb", "txt2", "20", "DDD"},
+		},
+	}}
+
+	for _, tcases := range testcases {
+		execStatements(t, []string{tcases.input})
+		output := qh.Expect(tcases.output)
+		expectNontxQueries(t, output)
+		time.Sleep(1 * time.Second)
+		log.Flush()
+		if tcases.table != "" {
+			expectData(t, tcases.table, tcases.data)
+		}
+	}
+	stats := globalStats.controllers[int32(vrId)].blpStats
+	require.Equal(t, 2, len(stats.PartialQueryCount.Counts()))
+	require.Equal(t, 2, len(stats.PartialQueryCacheSize.Counts()))
+	require.Equal(t, int64(2), stats.PartialQueryCacheSize.Counts()["insert"])
+	require.Equal(t, int64(3), stats.PartialQueryCount.Counts()["insert"])
+	require.Equal(t, int64(2), stats.PartialQueryCacheSize.Counts()["update"])
+	require.Equal(t, int64(4), stats.PartialQueryCount.Counts()["update"])
```

Comment:
IMO this would make more sense and be easier to build upon if each test case has info so that we could aggregate these in the for loop and check at the end. Each test case could have an optional boolean for partialInsert, partialUpdate, etc. and we increment a counter in the for loop when it's set.